builder/format.go

```diff
@@ -64,3 +68,63 @@ func RevertTrimmedGroupsINI(b *Builder) error {
 	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"))
 }
 
+func getLastBuildVersion(b *Builder) (string, error) {
+	// Likely the first build or some weirdness that requires us to get the container image and build it anyway
+	var lastVer []byte
+	var err error
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER")
+	if lastVer, err = ioutil.ReadFile(filename); err != nil {
+		return "", errors.Wrap(err, "Cannot find last built version, must get container image")
+	}
+	data := string(lastVer)
+	ver := strings.Split(data, "\n")
+
+	return ver[0], nil
+}
+
+// CheckBumpNeeded returns nil if it successfully deduces there is no format
+// bump boundary being crossed.
+func CheckBumpNeeded(b *Builder) (bool, error) {
+	version, err := getLastBuildVersion(b)
+	if err != nil {
+		return false, err
+	}
+	// Check what format our last built version is part of
+	versionURL := fmt.Sprintf("https://download.clearlinux.org/update/%s/format", version)
```

Comment:
You could rewrite this to use `b.getUpstreamFormatRange(version string)`, a) because it already exists and gets this value, and b) so you're not hardcoding the upstream url.

If you don't want use that (because it gets the other version values too and you're not using them), there's also the `b.DownloadFileFromUpstream` for times you need to grab individual files like this, which also lets you correctly use the `b.UpstreamURL` value automatically.

That said, `UpdateVersions` does the same conceptual check by doing one call to `format, first, latest, err := b.getUpstreamFormatRange(b.UpstreamVer)`, and then checking if the new format is different with `case nextUpstream < first, nextUpstream > latest:`, which could save you the second call.

Reply:
Sure, I can consolidate and use the other.